pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -94,7 +112,7 @@ in
     postInstall = ''
       PATH=$out/bin:$PATH patchShebangs $out
 
-      ${optionalString enableNpm ''
+      ${optionalString (enableNpm && stdenv.hostPlatform == stdenv.buildPlatform) ''
```

Comment:
maybe the build-time npm can be used to create bash completions for cross compiling